### ISSUE TITLE
Faster covariance estimation for bundle adjustment problems

### ISSUE THREAD

**Issue by sandwichmaker** [ceres-solver/ceres-solver#280]

**Comment by sandwichmaker** [comment]
https://ceres-solver-review.googlesource.com/c/ceres-solver/+/9581

**Comment by v01d** [comment]
Hi, is this still being considered?

**Comment by sandwichmaker** [comment]
yes, but the CL needs to be completely re-written. I have not had time to work on it, but I'd be happy to review the code if you or someone else is willing to work on it.

**Comment by leCaTo** [comment]
In a way to make some contribution to the subject, there have been some recent works from Michal Polic et al. addressing rank deficiencies in SfM/VisualSLAM while computing covariances: [in ECCV2018](http://openaccess.thecvf.com/content_ECCV_2018/papers/Michal_Polic_Fast_and_Precise_ECCV_2018_paper.pdf) and [3DV2017](http://people.ciirc.cvut.cz/~policmic/files/Polic___3DV_2017_IEEE_format.pdf). There is [code available](https://michalpolic.github.io/usfm.github.io/) and he has compared performance against ceres's MP pseudoinverse.
